src/Umbraco.Infrastructure/Runtime/RuntimeState.cs

```diff
@@ -215,13 +215,27 @@ public void DetermineRuntimeLevel()
                         _logger.LogDebug("Could not connect to database.");
                     }
 
```

Comment:
This may not be 100% correct, but I wonder if something like the following will get us past the complexity check.

```
var canInstallMissingDatabase = _globalSettings.Value.InstallMissingDatabase;

if (!canInstallMissingDatabase)
{
    _logger.LogDebug("InstallMissingDatabase is false. Not creating new database...");
}

if ((canInstallMissingDatabase == null && _databaseProviderMetadata.CanForceCreateDatabase(_databaseFactory))
|| canInstallMissingDatabase)
{
    // ok to install on a configured but missing database
    Level = RuntimeLevel.BootFailed;
    Reason = RuntimeLevelReason.InstallMissingDatabase;
    return;
}
```